fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/logical/PushdownFilterThroughSetOperation.java

```diff
@@ -45,11 +45,11 @@ public class PushdownFilterThroughSetOperation extends OneRewriteRuleFactory {
 
     @Override
     public Rule build() {
-        return logicalFilter(logicalSetOperation()).then(filter -> {
-            LogicalSetOperation setOperation = filter.child();
+        return logicalFilter(logicalSetOperation()).when(f -> f.child().getQualifier() == Qualifier.ALL).then(f -> {
```

Comment:
why must ALL?

Reply:
Because Union will be `Agg + UnionAll`, We just need match `UnionAll` is enough

Reply:
![image](https://github.com/apache/doris/assets/30525741/4228586d-ccac-4eee-8b3a-1d91adaea96e)

original code also didn't handle it